dataactvalidator/config/sqlrules/b11_object_class_program_activity_2.sql

```diff
@@ -1,7 +1,35 @@
+CREATE OR REPLACE function pg_temp.is_zero(numeric) returns integer AS $$
+BEGIN
+    perform CAST($1 AS numeric);
+    CASE WHEN $1 <> 0
+        THEN return 1;
+        ELSE return 0;
+    END CASE;
+EXCEPTION WHEN others THEN
+    return 0;
+END;
+$$ LANGUAGE plpgsql;
+
 SELECT
     op.row_number,
     op.object_class
 FROM object_class_program_activity AS op
 WHERE op.submission_id = {}
-AND op.object_class IN ('000', '00', '0')
+AND op.object_class IN ('0000', '000', '00', '0')
+AND pg_temp.is_zero(op.deobligations_recov_by_pro_cpe) + pg_temp.is_zero(op.gross_outlay_amount_by_pro_cpe) +
+    pg_temp.is_zero(op.gross_outlay_amount_by_pro_fyb) + pg_temp.is_zero(op.gross_outlays_delivered_or_cpe) +
```

Comment:
You mean instead of doing `is_zero`? They can potentially have negative numbers I think. Adding together the positives and negatives, in rare cases, might result in 0.